### PR TITLE
fix: prevent reuse mutated route option for head

### PR DESCRIPTION
**Pull request by climba03003**
Fixes: #4271 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — CHANGES_REQUESTED** [review]
Can you add a test?

File: test/route.test.js [hunk 1]
```diff
@@ -1464,3 +1464,32 @@ test('invalid url attribute - non string URL', t => {
...
+test('exposeHeadRoute should not reuse the same route option', async t => {
+  t.plan(2)
+
+  const fastify = Fastify()
+
+  // we update the onRequest hook in onRoute hook
+  // if we reuse the same route option
+  // that means we will append another function inside the array
+  fastify.addHook('onRoute', function (routeOption) {
+    if (Array.isArray(routeOption.onRequset)) {
```
Comment (Uzlopak) [inline test/route.test.js:1477]: typo:
```suggestion
    if (Array.isArray(routeOption.onRequest)) {
```

Comment (climba03003) [reply test/route.test.js]: Typo in all place

**Comment by Uzlopak** [comment]
@mcollina 
If we can get this merged and released I can close in fastify-rate-limit https://github.com/fastify/fastify-rate-limit/pull/267

maybe not a strong argument to make another release but i like when PRs are closed :D

**Review by Uzlopak — APPROVED** [review]
I approve it because it is fixing the issue i have in fastify-rate-limit

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**